be/src/vec/exec/format/parquet/vparquet_group_reader.cpp

```diff
@@ -24,235 +24,57 @@
-                    conjuncts.emplace_back(conjunct_ctxs[conj_idx]);
-                }
-            }
-        }
-    }
-}
-
-Status RowGroupReader::_init_column_readers() {
+Status RowGroupReader::_init_column_readers(const FieldDescriptor& schema,
+                                            std::vector<RowRange>& row_ranges) {
     for (auto& read_col : _read_columns) {
         SlotDescriptor* slot_desc = read_col.slot_desc;
-        FieldDescriptor schema = _file_metadata->schema();
         TypeDescriptor col_type = slot_desc->type();
-        const auto& field = schema.get_column(slot_desc->col_name());
-        const tparquet::RowGroup row_group =
-                _file_metadata->to_thrift_metadata().row_groups[_current_row_group];
-        ParquetColumnReader* reader = nullptr;
-        RETURN_IF_ERROR(ParquetColumnReader::create(_file_reader, MAX_PARQUET_BLOCK_SIZE, field,
-                                                    read_col, slot_desc->type(), row_group,
-                                                    reader));
+        auto field = const_cast<FieldSchema*>(schema.get_column(slot_desc->col_name()));
+        VLOG_DEBUG << "field: " << field->debug_string();
+        std::unique_ptr<ParquetColumnReader> reader;
+        RETURN_IF_ERROR(ParquetColumnReader::create(_file_reader, field, read_col, _row_group_meta,
+                                                    row_ranges, reader));
         if (reader == nullptr) {
+            VLOG_DEBUG << "Init row group reader failed";
             return Status::Corruption("Init row group reader failed");
         }
-        _column_readers[slot_desc->id()] = reader;
+        _column_readers[slot_desc->id()] = std::move(reader);
     }
     return Status::OK();
 }
 
-Status RowGroupReader::fill_columns_data(Block* block, const int32_t group_id) {
-    // get ColumnWithTypeAndName from src_block
+Status RowGroupReader::next_batch(Block* block, size_t batch_size, bool* _batch_eof) {
+    if (_read_rows >= _total_rows) {
+        *_batch_eof = true;
+    }
     for (auto& read_col : _read_columns) {
-        const tparquet::RowGroup row_group =
-                _file_metadata->to_thrift_metadata().row_groups[_current_row_group];
-        auto& column_with_type_and_name = block->get_by_name(read_col.slot_desc->col_name());
-        RETURN_IF_ERROR(_column_readers[read_col.slot_desc->id()]->read_column_data(
-                row_group, &column_with_type_and_name.column));
-        VLOG_DEBUG << column_with_type_and_name.name;
+        auto slot_desc = read_col.slot_desc;
+        auto& column_with_type_and_name = block->get_by_name(slot_desc->col_name());
+        auto column_ptr = column_with_type_and_name.column;
+        auto column_type = column_with_type_and_name.type;
+        RETURN_IF_ERROR(_column_readers[slot_desc->id()]->read_column_data(
+                column_ptr, column_type, batch_size, &_read_rows, _batch_eof));
```

Comment:
`_read_rows` should be a cumulative value. But here it just be set to the number values of current batch.